src/main/java/de/tum/in/www1/artemis/repository/StatisticsRepository.java

```diff
@@ -114,4 +205,343 @@
+     * and value the number of users in this interval
+     *
+     * @param span DAY,WEEK,MONTH or YEAR
+     * @param result the result given by the Repository call
+     * @param startDate the startDate of the period
+     * @param graphType the graphType for which the List should be converted
+     * @return A List<Map<String, Object>> analogue to other database calls
+     */
+    private List<Map<String, Object>> convertMapList(SpanType span, List<Map<String, Object>> result, ZonedDateTime startDate, GraphType graphType) {
+        Map<Object, List<String>> users = new HashMap<>();
+        for (Map<String, Object> listElement : result) {
+            Object index;
+            ZonedDateTime date;
+            if (graphType == GraphType.LOGGED_IN_USERS) {
+                Instant instant = (Instant) listElement.get("day");
+                date = instant.atZone(startDate.getZone());
+            }
+            else {
+                date = (ZonedDateTime) listElement.get("day");
+            }
+            if (span == SpanType.DAY) {
+                index = date.getHour();
+            }
+            else if (span == SpanType.WEEK || span == SpanType.MONTH) {
+                index = date.getDayOfMonth();
+            }
+            else if (span == SpanType.QUARTER) {
+                index = getWeekOfDate(date);
+            }
+            else {
+                index = date.getMonth();
+            }
+            String username = listElement.get("username").toString();
+            List<String> usersInSameSlot = users.get(index);
+            // if this index is not yet existing in users
+            if (usersInSameSlot == null) {
+                usersInSameSlot = new ArrayList<>();
+                usersInSameSlot.add(username);
+                users.put(index, usersInSameSlot);
+            }   // if the value of the map for this index does not contain this username
+            else if (!usersInSameSlot.contains(username)) {
+                usersInSameSlot.add(username);
+            }
+        }
+        return fillMapList(users, span, startDate);
+    }
+
+    /**
+     * Helper class for the ConvertMapList method, which takes the users in the same timeslot as well as some attributed needed
+     * for calculation to convert these into a Map List which is then returned
+     *
+     * @param users the List of Maps each containing the date and a list of users
+     * @param span the spanType for which we created the users List
+     * @param startDate the startDate which we need for mapping into timeslots
+     * @return A List<Map<String, Object>> with no duplicated user per timeslot
+     */
+    private List<Map<String, Object>> fillMapList(Map<Object, List<String>> users, SpanType span, ZonedDateTime startDate) {
+        List<Map<String, Object>> returnList = new ArrayList<>();
+        users.forEach((k, v) -> {
```

Comment:
why are you still using one letter variables? I really get angry now! I explained this extensively in the last meetings that we want to void this. Is this so difficult? :-(

Reply:
Sorry, was still old code. renamed it to a meaningful variable